wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -573,6 +586,49 @@ private Optional<ExpressionFactory<String>> getExpressionFactory(RelationalTranf
     return ctx.getEngine().getExpressionFactory(StringExpressionFactoryType.SQL);
   }
 
+  /**
+   * This method emits all metrics for the given list of directives
+   *
+   * @param directives a list of Wrangler directives
+   * @param metrics CDAP {@link Metrics} object using which metrics can be emitted
+   * @throws DirectiveLoadException
+   */
+  private void emitDirectiveMetrics(List<Directive> directives, Metrics metrics) throws DirectiveLoadException {
+    for (Directive directive : directives) {
+      List<EntityMetricDef> entityMetricDefs = new ArrayList<>();
+
+      // Emit directive usage metric only if directive is present in system directive registry
+      if (registry.get(Contexts.SYSTEM, directive.define().getDirectiveName()) != null) {
+        entityMetricDefs.add(getDirectiveUsageMetric(directive.define().getDirectiveName()));
+      }
+
+      if (directive instanceof EntityMetrics) {
```

Comment:
instanceof is a code smell and should be avoided when possible.

In this particular case, the Directive interface can be modified to have the method that is in EntityMetrics, with a default implementation that returns nothing.

Reply:
EntityMetrics is a common interface that can be implemented by any "entity" in wrangler that would like to emit metrics. We are treating a Directive as an entity here. Hence why the function shouldn't be moved specifically to the Directive interface.